tests/api/python/tests.py.in

```diff
@@ -263,7 +264,56 @@ class TestSTP(unittest.TestCase):
             s.check(c == op(a))
             self.assertEqual(s.model()['c'], op(A) % 2**32)
 
+    def _test_solver(self, solver, is_default_solver=False):
+        """
+        Helper method to validate that the passed in "solver" can be set and
+        used
+        """
+        supports_method = getattr(self.s, "supports{:s}".format(solver))
+        use_method = getattr(self.s, "use{:s}".format(solver))
+        is_using_method = getattr(self.s, "isUsing{:s}".format(solver))
+
+        supported = supports_method()
+
+        if is_default_solver:
+            self.assertTrue(supported)
+
+        #
+        # If the solver is supported, the "use" and "isUsing" methods should
+        # return true -- similarly, if the solver is not supported, then they
+        # both return false.
+        #
+        # We can use that as the condition for the assertion
+        #
+        self.assertTrue(use_method() == supported)
+        self.assertTrue(is_using_method() == supported)
```

Comment:
If one solver is selected, do other solvers get disabled? If so, we should test for that, maybe.

Reply:
Please use `self.assertEqual()` or the appropriate assertion method here.

Reply:
I extended the test to make sure no other solvers were in use, and changed to `assertEqual`.